metric_config_parser/tests/test_analysis.py

```diff
@@ -568,3 +572,131 @@ def test__merge_param_raises(self, input):
+
+            [metrics.spam_ham]
+            depends_on = ["spam", "ham"]
+
+            [metrics.spam_ham.statistics.bootstrap_mean]
+            """
+        )
+
+        spec = AnalysisSpec.from_dict(toml.loads(config_str))
+        cfg = spec.resolve(experiments[0], config_collection)
+
+        assert len(cfg.metrics[AnalysisPeriod.WEEK]) == 3
+
+        metric = [m for m in cfg.metrics[AnalysisPeriod.WEEK] if m.metric.name == "spam_ham"][
+            0
+        ].metric
+        assert metric.select_expression is None
+        assert metric.data_source is None
+        assert len(metric.depends_on) == 2
+
+        upstream_metrics = [m.metric.name for m in metric.depends_on]
+        assert "spam" in upstream_metrics
+        assert "ham" in upstream_metrics
+
+    def test_depends_on_definition(self, experiments, config_collection):
+        config_str = dedent(
+            """
+            [metrics]
+            weekly = ["active_hours", "view_about_logins", "combined_metric"]
+
+            [metrics.combined_metric]
+            depends_on = ["active_hours", "view_about_logins"]
+
+            [metrics.combined_metric.statistics.bootstrap_mean]
+            """
+        )
+
+        spec = AnalysisSpec.from_dict(toml.loads(config_str))
+        cfg = spec.resolve(experiments[0], config_collection)
+
+        assert len(cfg.metrics[AnalysisPeriod.WEEK]) == 3
+
+        metric = [
+            m for m in cfg.metrics[AnalysisPeriod.WEEK] if m.metric.name == "combined_metric"
+        ][0].metric
+        assert metric.select_expression is None
+        assert metric.data_source is None
+        assert len(metric.depends_on) == 2
+
+        upstream_metrics = [m.metric.name for m in metric.depends_on]
+        assert "active_hours" in upstream_metrics
+        assert "view_about_logins" in upstream_metrics
+
+    def test_non_existing_depends_on(self, experiments, config_collection):
+        config_str = dedent(
+            """
+            [metrics]
+            weekly = ["combined_metric"]
+
+            [metrics.combined_metric]
+            depends_on = ["non_existing"]
+
+            [metrics.combined_metric.statistics.bootstrap_mean]
+            """
+        )
+
+        spec = AnalysisSpec.from_dict(toml.loads(config_str))
+
+        with pytest.raises(DefinitionNotFound):
+            spec.resolve(experiments[0], config_collection)
+
+    def test_empty_metric_definition_not_allowed(self, experiments, config_collection):
+        config_str = dedent(
+            """
+            [metrics]
+            weekly = ["empty_metric"]
+
+            [metrics.empty_metric]
+
+            [metrics.empty_metric.statistics.bootstrap_mean]
+            """
+        )
+
+        spec = AnalysisSpec.from_dict(toml.loads(config_str))
+
+        with pytest.raises(DefinitionNotFound):
+            spec.resolve(experiments[0], config_collection)
+
+    def test_circular_depends_on(self, experiments, config_collection):
```

Comment:
Should we have a test for multiple nesting? Something like

```python
        config_str = dedent(
            """
            [metrics]
            weekly = ["spam", "ham", "wham"]

            [metrics.spam]
            data_source = "main"
            select_expression = "1"

            [metrics.spam.statistics.bootstrap_mean]

            [metrics.ham]
            depends_on = ["spam"]

            [metrics.ham.statistics.bootstrap_mean]

            [metrics.wham]
            depends_on = ["ham"]

            [metrics.ham.statistics.bootstrap_mean]
            """
        )
```

Reply:
Added this as a test, and yeah, it is working